Framework/src/Triggers.cxx

```diff
@@ -109,9 +115,23 @@ TriggerFcn Never(const Activity& activity)
   };
 }
 
-TriggerFcn EndOfRun(const Activity&)
+TriggerFcn EndOfRun(const std::string& kafkaBrokers, const std::string& topic, const std::string& detector, const std::string& taskName, const Activity& activity)
 {
-  return NotImplemented("EndOfRun");
+  auto copiedActivity = activity;
+  auto poller = std::make_shared<core::KafkaPoller>(kafkaBrokers, createKafkaGroupId("EOR_postprocessing", detector, taskName));
+  poller->subscribe(topic);
+  return [poller, copiedActivity]() mutable -> Trigger {
+    for (const auto& record : poller->poll()) {
+      if (auto event = proto::recordToEvent(record.value())) {
+        if (proto::end_of_run::check(*event, copiedActivity.mProvenance, copiedActivity.mId)) {
+          auto newActivityForTrigger = copiedActivity;
+          proto::end_of_run::fillActivity(*event, newActivityForTrigger);
```

Comment:
I don't think we should take any fields from the Activity of the postprocessing task. After all, we want to know about other ECS environment starting/finishing and any metadata we can extract about it.

Reply:
Right now it is setup that if Activity has empty fields, it is not filled and trigger matches against everything. And if you want to provide some filter over runnumber/envid, than it is possible.

Reply:
uhm, but `Activity` received by argument of this function is set as the current Activity (i.e. current run, current env ID if it's ran with ECS), so user won't always have control over that. I think we can safely assume that the user should get any available SOR/EOR except of their own.

Reply:
How? When I look into triggerFactory, than this activity is from config

Reply:
And this is the config that is passed: https://github.com/AliceO2Group/QualityControl/blob/f6820b593ec6c69eff455bb2151d19501f65ec79/Framework/src/PostProcessingRunner.cxx#L196

You can see that the `activityFromDriver` is used.

Reply:
or is the config always templated to the current activity? And if so, than what should I do, just invert the filtering? return anything except id if current run?

Reply:
yes, return a valid trigger if any other run starts or ends, but not this one.